### ISSUE TITLE
register viewpoint implicitly commits session but maybe it shouldn't?

### ISSUE THREAD

**Issue by kuba-lilz** [Flask-Middleware/flask-security#666]
Changelog for version `5.0.0` states that user registration action should now be committed by application

```
A very old piece of code in registrable, would immediately commit to the DB when a new user was created. It is now consistent with all other views, and has the caller responsible for committing the transaction - usually by setting up a flask after_this_request action. This could affect an application that captured the registration signal and stored the user object for later use - this user object would likely be invalid after the request is finished.
```

I was happy to read this, since it gives me control over registration logic I don't have with 4.x.
However, testing this change with data store based on `flask_security.datastore.SQLAlchemySessionUserDatastore` it turns out that even if I have no explicit commit in my `datastore.create_user` (it only constructs user object and calls `self.put(user)`), user object still gets committed to datastore.

Reading through the code I think the reason is that `views.py::register` includes `after_this_request(view_commit)`
```python
    if form.validate_on_submit():
        after_this_request(view_commit)
        did_login = False
        user = register_user(form)
```
around lines 307~308

 https://github.com/Flask-Middleware/flask-security/blob/01d2a33d67f75ce3607a22d6eef3367e1f7a21e8/flask_security/views.py#L307

 `view_commit` basically calls `session.commit()`.

The end result is that execution of `register` endpoint triggers `session.commit()` implicitly.
Is this as intended? Personally I hope that not really, and will be changed, because I would really like to have more control over what gets committed and when in registration endpoint ^^;

**Comment by kuba-lilz** [comment]
After reading the documentation carefully, I realized everything is in order with it.
`register()` endpoint is the caller of `register_user / datastore.create_user`, and it commits the session afterwards - just as documentation specifies.

Maybe not exactly what I personally would hope for, but in line with documentation nonetheless, so the concern I had in the issue is wrong. My apologies.

**Comment by jwag956** [comment]
I think Flask-Security really has to commit - otherwise, at the end of the request, the DB transaction will be lost (I believe).
However - with the change in 5.0 - the commit() happens at the very end of the request - so for example you can use the user.registered signal to make any changes/additions to the User Model well before it ever gets committed.

**Comment by kuba-lilz** [comment]
Thank you for the comment!

I understand someone has to commit the change at some point - so either Flask Security does it for me, or offers me a hook that in most cases would do the same.

I made this issue because frankly I jumped to conclusions a bit too fast after reading about the change in 5.0.0, my apologies.
And moving session commit after all signals are executed is a nice improvement over previous versions.

Thinking more about why I want to have control over session commit, it's much more to do with being limited in what we can do in validation now - I described the problem in more generic terms and proposed solution in #667
